common/wireaddr.h

```diff
@@ -37,6 +37,14 @@ struct wireaddr {
 	u16 port;
 };
 
+/* We use wireaddr to tell gossipd both what to listen on, and what to
+ * announce */
+enum addr_listen_announce {
+	ADDR_LISTEN = 1,
+	ADDR_ANNOUNCE = 2,
+	ADDR_LISTEN_AND_ANNOUNCE = 3
```

Comment:
We take advantage that `ADDR_LISTEN_AND_ANNOUNCE` == `ADDR_LISTEN | ADDR_ANNOUNCE`.  Maybe it would be better to be explicit about that here?

    enum addr_listen_announce {
        ADDR_LISTEN = 1 << 0, // be explicit that we are using bits
        ADDR_ANNOUNCE = 1 << 1,
        ADDR_LISTEN_AND_ANNOUNCE = ADDR_LISTEN | ADDR_ANNOUNCE // be explicit that this is combination of bits.
    };

Otherwise it may naively seem that numbers are just incrementing.